airbyte-container-orchestrator/src/main/java/io/airbyte/container_orchestrator/ReplicationJobOrchestrator.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.container_orchestrator;
+
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.config.Configs;
+import io.airbyte.config.ReplicationOutput;
+import io.airbyte.config.StandardSyncInput;
+import io.airbyte.scheduler.models.IntegrationLauncherConfig;
+import io.airbyte.scheduler.models.JobRunConfig;
+import io.airbyte.workers.DefaultReplicationWorker;
+import io.airbyte.workers.ReplicationWorker;
+import io.airbyte.workers.WorkerConfigs;
+import io.airbyte.workers.WorkerConstants;
+import io.airbyte.workers.WorkerUtils;
+import io.airbyte.workers.process.AirbyteIntegrationLauncher;
+import io.airbyte.workers.process.IntegrationLauncher;
+import io.airbyte.workers.process.ProcessFactory;
+import io.airbyte.workers.protocols.airbyte.AirbyteMessageTracker;
+import io.airbyte.workers.protocols.airbyte.AirbyteSource;
+import io.airbyte.workers.protocols.airbyte.DefaultAirbyteDestination;
+import io.airbyte.workers.protocols.airbyte.DefaultAirbyteSource;
+import io.airbyte.workers.protocols.airbyte.EmptyAirbyteSource;
+import io.airbyte.workers.protocols.airbyte.NamespacingMapper;
+import io.airbyte.workers.temporal.sync.ReplicationLauncherWorker;
+import java.nio.file.Path;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class ReplicationJobOrchestrator implements JobOrchestrator<StandardSyncInput> {
+
+  private final ProcessFactory processFactory;
+  private final WorkerConfigs workerConfigs;
+  private final Configs configs;
+
+  public ReplicationJobOrchestrator(final Configs configs, final WorkerConfigs workerConfigs, final ProcessFactory processFactory) {
+    this.configs = configs;
+    this.workerConfigs = workerConfigs;
+    this.processFactory = processFactory;
+  }
+
+  @Override
+  public String getOrchestratorName() {
+    return "Replication";
+  }
+
+  @Override
+  public Class<StandardSyncInput> getInputClass() {
+    return StandardSyncInput.class;
+  }
+
+  @Override
+  public void runJob() throws Exception {
+    final JobRunConfig jobRunConfig = readJobRunConfig();
```

Comment:
to make sure I understand - how does this interact with the `SyncWorkflowImpl`?

Reply:
So the sync workflow calls an orchestrator launcher activity, which launches a pod/container running the orchestrator itself (the `ContainerOrchestratorApp` which runs the`ReplicationJobOrchestrator`.

Reply:
got it, so the control flow = user triggers job -> temporal sync workflow picks it up -> pod created running ContainerOrchestratorApp -> runs replication job -> this creates the source and destination pods. is that right?

Reply:
yep